### PR TITLE
New Resource: `azurerm_api_management_diagnostic `

### PR DESCRIPTION
**Pull request by aqche**
Partially addresses: #4760

Adds the `azurerm_api_management_diagnostic` resource to the extent that the `2018-01-01` version of the `apimanagement` service allows. As noted in my comment on the feature request, I realize this doesn't have near the amount of options as the `2019-01-01` version would, but perhaps it can provide incremental value/serve as a step to addressing the issue. That being said, this is my first attempt at contributing a new resource to the provider so please let me know if you have any feedback! :)

```
--- PASS: TestAccAzureRMApiManagementDiagnostic_basic (2122.14s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm       2122.193s
```

```
--- PASS: TestAccAzureRMApiManagementDiagnostic_requiresImport (2911.42s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm       2911.465s
```

### CODE REVIEW

File: azurerm/resource_arm_api_management_diagnostic.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"identifier": {
+				Type:     schema.TypeString,
+				Required: true,
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_api_management_diagnostic.go]: presumably this is also:

```suggestion
				Required: true,
				ForceNew: true,
```

File: website/docs/r/api_management_diagnostic.html.markdown [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  api_management_name = "${azurerm_api_management.test.name}"
+  enabled             = true
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `identifier` - (Required) The diagnostic identifier for the API Management Service. Allowed values are `applicationinsights`.
```
Comment (tombuildsstuff) [inline website/docs/r/api_management_diagnostic.html.markdown]: Could we make this:

```suggestion
* `identifier` - (Required) The diagnostic identifier for the API Management Service. At this time the only supported value is `applicationinsights`. Changing this forces a new resource to be created.
```

File: website/docs/r/api_management_diagnostic.html.markdown [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The ID of the API Management Diagnostic.
+
+## Import
+
+API Management Diagnostics can be imported using their `resource id`, e.g.
```
Comment (tombuildsstuff) [inline website/docs/r/api_management_diagnostic.html.markdown]: to match this with other resources could we make this:

```suggestion
API Management Diagnostics can be imported using the `resource id`, e.g.
```

**Comment by tombuildsstuff** [comment]
@aqche we're preparing to start some fairly large-scale refactoring to the codebase - as such I'm going to push a commit to fix the issues above and rebase this, so that we can get this merged before that happens; I hope you don't mind!

**Comment by tombuildsstuff** [comment]
Tests pass:

Screenshot 2019-12-18 at 22 35 57

**Comment by aqche** [comment]
@tombuildsstuff thanks for the review and taking the time to make the changes as well! 🙏

**Pull request merged**